src/vals/symbolic_int.rs

```diff
@@ -0,0 +1,149 @@
+use crate::memory::{SymbolicIndexIndexVal, SymbolicIndexMemVal};
+use std::{
+    num::TryFromIntError,
+    ops::{Add, Sub},
+};
+
+use super::machine_eq::MachineEq;
+
+pub type Wraps = i64;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum SymbolicInt {
+    // Concrete
+    C(Wraps),
+    // Symbolic
+    S(Inner),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Inner {
+    Sym(String),
+    Add(Box<SymbolicInt>, Box<SymbolicInt>),
+    Sub(Box<SymbolicInt>, Box<SymbolicInt>),
+    Ite(EqCheck, Box<SymbolicInt>, Box<SymbolicInt>),
+    // The result of a read from a series of writes
+    RW(Box<(Vec<(SymbolicInt, SymbolicInt)>, SymbolicInt)>),
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct EqCheck {
+    l: Box<SymbolicInt>,
+    r: Box<SymbolicInt>,
+}
```

Comment:
The datatype holding a symbolic integer in the machine. A lot of this functionality can probably more abstracted over such as in `instructions/val.rs` but I think it makes sense to just build it out specific to an integer for now.

See here for how it gets converted into the z3 Int
https://github.com/WilfredTA/symbolic-stack-machines/blob/aec8aa99c2c0c19370f8402c3976a41300ed8f70/src/symbolic_int.rs#L43-L114